packages/compat/src/resolver-transform.ts

```diff
@@ -320,3 +328,16 @@ function handleComponentHelper(
 
   resolver.resolveComponentHelper(locator, moduleName, param.loc, impliedBecause);
 }
+
+function handleDynamicHelper(param: ASTv1.Node, resolver: Resolver, moduleName: string): void {
+  switch (param.type) {
+    case 'StringLiteral':
+      resolver.resolveDynamicHelper({ type: 'literal', path: param.value }, moduleName, param.loc);
+      break;
+    case 'TextNode':
+      resolver.resolveDynamicHelper({ type: 'literal', path: param.chars }, moduleName, param.loc);
+      break;
+    default:
+      resolver.resolveDynamicHelper({ type: 'other' }, moduleName, param.loc);
```

Comment:
@ef4 Do we actually want to handle this case? The contextual helpers RFCs mentions that passing a helper reference into the helper should be supported so that extra arguments can be added: https://emberjs.github.io/rfcs/0432-contextual-helpers.html 

I left it out in the draft PR since [Ember itself verifies that the helper isn't dynamic](https://github.com/emberjs/ember.js/blob/bfaed5a9a944bf31c99f5772da4f7681b7004555/packages/ember-template-compiler/lib/plugins/transform-resolutions.ts#L34). I think Embroider only needs to do anything if users pass in a helper name?

I think the code snippet in the RFC would throw an error with the current implementation?
```hbs
{{#let (helper "join-words" separator=",") as |join|}}
  {{#let (helper join "foo") as |foo|}}
    {{#let (helper foo "bar") as |foo-bar|}}
      {{foo-bar "baz"}}
    {{/let}}
  {{/let}}
{{/let}}
```

Reply:
Yes, this also generates an error, it's just done from the resolver instead of directly here because the resolver keeps track of all the errors.

Reply:
But I think it _shouldn't_ throw an error for that code snippet? Passing in a helper reference should work, according to the RFC, but with the currently merged implementation it will throw an error I think?

Unless I misunderstood the code, I think at the moment, errors will be thrown under Embroider while it works as expected in a classic build.